OpenRA.Game/Map/Map.cs

```diff
@@ -716,8 +722,21 @@ public byte[] SavePreview()
 						{
 							// Cell contains terrain
 							var type = tileset.GetTileInfo(Tiles[uv]);
-							leftColor = type != null ? type.LeftColor : Color.Black;
-							rightColor = type != null ? type.RightColor : Color.Black;
+							if (type != null)
+							{
+								if (useHeightModifiers)
+								{
+									var left = Exts.ColorLerp(Game.CosmeticRandom.NextFloat(), type.MinColor, type.MaxColor);
+									var right = Exts.ColorLerp(Game.CosmeticRandom.NextFloat(), type.MinColor, type.MaxColor);
+									var scale = float2.Lerp(minHeightBrightness, maxHeightBrightness, heightModifier * Height[uv]);
+									leftColor = Color.FromArgb((int)(scale * left.R).Clamp(0, 255), (int)(scale * left.G).Clamp(0, 255), (int)(scale * left.B).Clamp(0, 255));
+									rightColor = Color.FromArgb((int)(scale * right.R).Clamp(0, 255), (int)(scale * right.G).Clamp(0, 255), (int)(scale * right.B).Clamp(0, 255));
+								}
+								else
+									leftColor = rightColor = type.MinColor;
+							}
+							else
+								leftColor = rightColor = Color.Black;
```

Comment:
Shouldn't this be

```suggestion
								else
								{
									leftColor = type.MinColor;
									rightColor = type.MaxColor;
								}
							}
							else
								leftColor = rightColor = Color.Black;
```

?

Reply:
No. The two color values do not correspond to the left and right pixels in the isometric minimaps - this was bogus in the previous implementation.